fedora/client/bodhi.py

```diff
@@ -24,22 +24,376 @@
+        :kwarg rows_per_page: Limit the results to a certain number of rows per
+            page (min:1 max: 100)
+        :kwarg page: Return a specific page of results
+
+        """
+        if 'limit' in kwargs:  # bodhi1 compat
+            kwargs['rows_per_page'] = kwargs['limit']
+            del(kwargs['limit'])
+        return self.send_request('updates', verb='GET', params=kwargs)
+
+    def comment(self, update, comment, karma=0, email=None):
+        """ Add a comment to an update.
+
+        :arg update: The title of the update comment on.
+        :arg comment: The text of the comment.
+        :kwarg karma: The karma of this comment (-1, 0, 1)
+        :kwarg email: Whether or not to trigger email notifications
+
+        """
+        return self.send_request('comments/', verb='POST', auth=True,
+                data={'update': update, 'text': comment,
+                      'karma': karma, 'email': email,
+                      'csrf_token': self.csrf()})
+
+    def csrf(self, **kwargs):
+        return self.send_request('csrf', verb='GET',
+                                 params=kwargs)['csrf_token']
+
+    def parse_file(self, input_file):
+        """ Parse an update template file.
+
+        :arg input_file: The filename of the update template.
+
+        Returns an array of dictionaries of parsed update values which
+        can be directly passed to the ``save`` method.
+
+        """
+        from ConfigParser import SafeConfigParser
+
+        if not os.path.exists(input_file):
+            raise ValueError("No such file or directory: %s" % input_file)
+
+        config = SafeConfigParser()
+        read = config.read(input_file)
+
+        if len(read) != 1 or read[0] != input_file:
+            raise ValueError("Invalid input file: %s" % input_file)
+
+        updates = []
+
+        for section in config.sections():
+            update = {
+                'builds': section, 'bugs': config.get(section, 'bugs'),
+                'close_bugs': config.getboolean(section, 'close_bugs'),
+                'type': config.get(section, 'type'),
+                'request': config.get(section, 'request'),
+                'severity': config.get(section, 'severity'),
+                'notes': config.get(section, 'notes'),
+                'autokarma': config.get(section, 'autokarma'),
+                'stable_karma': config.get(section, 'stable_karma'),
+                'unstable_karma': config.get(section, 'unstable_karma'),
+                'suggest': config.get(section, 'suggest'),
+                }
+
+            updates.append(update)
+
+        return updates
+
+    def latest_builds(self, package):
+        return self.send_request('latest_builds', params={'package': package})
+
+    def testable(self):
+        """ Get a list of installed testing updates.
+
+        This method is a generate that yields packages that you currently
+        have installed that you have yet to test and provide feedback for.
+
```

Comment:
Might want to add a sentence to this docstring indicating that this function won't work on a system without dnf.